File: dotnet/src/Connectors/Connectors.Memory.Weaviate/Diagnostics/Verify.cs
// Copyright (c) Microsoft. All rights reserved.

using System;
using System.Diagnostics.CodeAnalysis;
using System.Runtime.CompilerServices;

namespace Microsoft.SemanticKernel.Connectors.Memory.Weaviate.Diagnostics;

internal static class Verify
{
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    internal static void NotNull([NotNull] object? obj, string message)
    {
        if (obj != null) { return; }

        throw new ArgumentNullException(null, message);
    }

    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    internal static void NotNullOrEmpty([NotNull] string? str, string message)
    {
        NotNull(str, message);
        if (!string.IsNullOrWhiteSpace(str)) { return; }

        throw new ArgumentOutOfRangeException(message);
    }

    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    internal static void ArgNotNullOrEmpty([NotNull] string? str, string paramName, [CallerMemberName] string? caller = default)
    {
        NotNull(str, paramName);
        if (!string.IsNullOrWhiteSpace(str)) { return; }

        throw new ArgumentException(paramName, $"Parameter {paramName} cannot be empty." + (!string.IsNullOrEmpty(caller) ? $"({caller})" : string.Empty));
    }
}

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/Verify.java
// Copyright (c) Microsoft. All rights reserved.
package com.microsoft.semantickernel;

import javax.annotation.Nullable;

public class Verify {
<<<<<<< AI

=======
<<<<<<< HEAD

=======
>>>>>>> beeed7b7a795d8c989165740de6ddb21aeacbb6f
>>>>>>> main
    public static void notNull(Object object) {
        assert object != null;
    }

    public static boolean isNullOrEmpty(@Nullable String s) {
        return s == null || s.isEmpty();
    }
<<<<<<< AI
=======
<<<<<<< HEAD
>>>>>>> main

    public static boolean isNullOrWhiteSpace(String s) {
        return s == null || s.matches("^\\w+$");
    }
<<<<<<< AI
=======
=======
>>>>>>> beeed7b7a795d8c989165740de6ddb21aeacbb6f
>>>>>>> main
}
